zipline/testing/core.py

```diff
@@ -1381,3 +1449,39 @@ def patched_read_csv(filepath_or_buffer, *args, **kwargs):
 
     with patch.object(module, 'read_csv', patched_read_csv):
         yield
+
+
+####################################
+# Shared factors for pipeline tests.
+####################################
+
+class AssetID(CustomFactor):
+    """
+    CustomFactor that returns the AssetID of each asset.
+
+    Useful for providing a Factor that produces a different value for each
+    asset.
+    """
+    window_length = 1
+    # HACK: We currently decide whether to load or compute a Term based on the
```

Comment:
This isn't true anymore.  We should be able to do `inputs=()` and remove `close` from the compute signature now.